Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -285,16 +311,41 @@ void OpenALStream::SoundLoop()
 				{
 					alBufferData(uiBufferTemp[iBuffersFilled], AL_FORMAT_51CHN32, dpl2, nSamples * FRAME_SURROUND_FLOAT, ulFrequency);
 				}
+				else if (fixed32_capable)
+				{
+					int surround_int32[OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS];
+
+					for (u32 i = 0; i < nSamples * SURROUND_CHANNELS; ++i)
+					{
+						// For some reason the DPL2 decoder outputs samples bigger than 1. Most are close to 2.5 and some go up to 8. How to deal with this? Hard clamping here, we need to fix the decoder or implement a limiter.
+						if (dpl2[i] >= 1)
+							surround_int32[i] = INT_MAX;
+						else if (dpl2[i] <= -1)
+							surround_int32[i] = INT_MIN;
+						else
+							surround_int32[i] = (int)((float)dpl2[i] * (1 << 31));
```

Comment:
Thanks, that makes sense and should work :)
If someone with a X-Fi can test it please do so. 
I will amend the commit again with 1LL.